railties/lib/rails/commands/server/server_command.rb

```diff
@@ -44,17 +71,24 @@ def set_environment
     end
 
     def start
+      missing_handler = false
+
       print_boot_information
       trap(:INT) { exit }
       create_tmp_directories
       setup_dev_caching
       log_to_stdout if options[:log_stdout]
 
       super
+    rescue LoadError
```

Comment:
We should make sure to only rescue a LoadError for `"rack/handler/#{options[:using]}"`; if something else is wrong, the original exception will do a better job of describing the issue.

Reply:
`Rack::Handler.get` can raise either `LoadError` or `NameError`, if the Puma constant does not exist after the `rack/handler/puma` require, for example. I think I can override the `Rack::Server#server` method to isolate the error catch.

Reply:
Can not deal with this by explicitly calling `Rack::Handler.get` before starting server? 
I am also concerned that other errors by `LoadError` will be wrapped by this.

Reply:
We can, I'll do the changes accordingly.